zk-rs/src/account.rs

```diff
@@ -0,0 +1,319 @@
+use ark_bn254::Fr;
+use crypto_box::SecretKey;
+use num_bigint::BigUint;
+use wasm_bindgen::prelude::wasm_bindgen;
+
+use crate::{
+    blake2::{blake2_hash, blake2_hash_str},
+    poseidon::poseidon_hash,
+};
+
+#[wasm_bindgen]
+pub struct Account {
+    // solana_public_key: Vec<u8>,
+    // seed: Vec<u8>,
+    public_key: Vec<u8>,
+    private_key: Vec<u8>,
+    encryption_public_key: Vec<u8>,
+    encryption_private_key: Vec<u8>,
+    aes_secret: String,
+    hashing_secret: String,
+}
+
+#[wasm_bindgen]
+impl Account {
+    #[wasm_bindgen(constructor)]
+    pub fn new(seed: String) -> Self {
+        // if seed.len() < 32 {
+        //     return Err(AccountError::new(
+        //         AccountErrorCode::InvalidSeedSize,
+        //         "constructor",
+        //         "seed too short length less than 32",
+        //     ));
+        // }
+        let encryption_private_key = Account::get_encryption_private_key(seed.clone());
```

Comment:
why not use references instead of cloning ```seed``` ?

Reply:
same for all the method calls below

Reply:
I removed account.rs from this branch for now, because it's not related to this PR and unfinished yet. 
I'll another PR with account.rs after proofgen.